packages/flutter/lib/src/widgets/interactive_viewer.dart

```diff
@@ -726,11 +565,141 @@ class _InteractiveViewerState extends State<InteractiveViewer> with TickerProvid
+        final ClosestPoints closestPoints = invalidSide.findClosestPointsRect(_boundaryRect);
+        final Offset difference = closestPoints.a - closestPoints.b;
+        if (magnitudesAligned) {
+          final bool xMisaligned = difference.dx != 0.0 && value.dx != 0.0
+              && (difference.dx > 0.0 != value.dx > 0.0);
+          final bool yMisaligned = difference.dy != 0.0 && value.dy != 0.0
+              && (difference.dy > 0.0 != value.dy > 0.0);
+          if (xMisaligned || yMisaligned) {
+            magnitudesAligned = false;
+          }
+        }
+        return Offset(
+          difference.dx.abs() > value.dx.abs() ? difference.dx : value.dx,
+          difference.dy.abs() > value.dy.abs() ? difference.dy : value.dy,
+        );
+      });
+    }
+
+    // If there is only one invalid side, or if there are multiple but they are
+    // invalid in different directions, then just try to correct the single most
+    // invalid side.
+    if (!magnitudesAligned || invalidSides.length == 1) {
+      // Find the side that is farthest from being in a valid position.
+      final LineSegment invalidSide = (invalidSides
+        ..sort((LineSegment a, LineSegment b) {
+          final ClosestPoints closestPointsA = a.findClosestPointsRect(_boundaryRect);
+          final double distanceA = _distanceBetweenPoints(
+            closestPointsA.a,
+            closestPointsA.b,
+          ).abs();
+          final ClosestPoints closestPointsB = b.findClosestPointsRect(_boundaryRect);
+          final double distanceB = _distanceBetweenPoints(
+            closestPointsB.a,
+            closestPointsB.b,
+          ).abs();
+          return distanceB.compareTo(distanceA);
+        })).first;
+
+      // Find the nearest points on boundaryRect and the viewportSide that doesn't
+      // intersect to each other, and move the points on top of each other.
+      final ClosestPoints closestPoints = invalidSide.findClosestPointsRect(_boundaryRect);
+      correction = closestPoints.a - closestPoints.b;
+    }
+
+    final Matrix4 correctedMatrix = matrix.clone()
+        ..translate(correction.dx, correction.dy);
+    final Quad correctedNextViewport = _transformViewport(correctedMatrix, _viewport);
+
+    final LineSegment correctedViewportSide01 = LineSegment.vector(
+      correctedNextViewport.point0,
+      correctedNextViewport.point1,
+    );
+    final LineSegment correctedViewportSide12 = LineSegment.vector(
+      correctedNextViewport.point1,
+      correctedNextViewport.point2,
+    );
+    final LineSegment correctedViewportSide23 = LineSegment.vector(
+      correctedNextViewport.point2,
+      correctedNextViewport.point3,
+    );
+    final LineSegment correctedViewportSide30 = LineSegment.vector(
+      correctedNextViewport.point3,
+      correctedNextViewport.point0,
+    );
+
+    // If this matrix is invalid and uncorrectable, then return defaultMatrix.
+    if (!correctedViewportSide01.intersectsRect(_boundaryRect)
```

Comment:
so for cases like this: https://jamboard.google.com/d/1l4Gg2AAs8d_8fGEAK_diDle-9qSQjYYFPkxpzUXAS84/viewer?f=1, this will try to correct the matrix by moving the child to the right edge of the viewport because it's farther, then return the default matrix because the top left corner of the child is inside the viewport?